sdf/1.7/frame.sdf

```diff
@@ -1,16 +1,32 @@
 <!-- Frame -->
 <element name="frame" required="*">
-  <description>A frame of reference to which a pose is relative.</description>
+  <description>A frame of reference in which poses may be expressed.</description>
 
   <attribute name="name" type="string" default="" required="1">
-    <description>Name of the frame. This name must not match another frame defined inside the parent that this frame is attached to.</description>
+    <description>
+      Name of the frame. It must be unique whithin its scope (model/world),
+      i.e., it must not match the name of another frame, link, joint, or model
+      within the same scope.
+    </description>
   </attribute>
 
   <attribute name="attached_to" type="string" default="" required="*">
     <description>
-      Name of the link or frame to which this frame is attached.
-      If a frame is specified, recursively following the attached_to attributes
-      of the specified frames must lead to the name of a link, a model, or the world frame.
+      If specified, this frame is attached to the specified frame. The specified
+      frame must be within the same scope and may be defined implicitly, i.e.,
+      the name of any //frame, //model, //joint, or //link within the same scope
+      may be used.
+
+      If missing, this frame is attached to the containing scope's frame. Within
+      a //world scope this is the implicit world frame, and within a //model
+      scope this is the implicit model frame.
+
+      A frame moves jointly with the frame it is @attached_to. This is different
+      from //pose/@relative_to. @attached_to defines how the frame is attached
+      to a //link, //model, or //world frame, while //pose/@relative_to defines
+      how the frame's pose is represented numerically. As a result, following
+      the chain of @attached_to attributes must always lead to a //link,
+      //model, //world, or //joint (implicitly attached_to its child //link). 
```

Comment:
I don't see the value of adding `//joint` here because that's not where the chain terminates. It terminates on the child link, which is covered by the previous portion of the sentence.

Reply:
This was added during the review by @scpeters : https://github.com/ignitionrobotics/sdformat/pull/666#discussion_r690636440

It adds value in the sense that `//joint` doesn't have a @attached_to attribute and thus you can't follow `//joint/@attached_to` to the child `//link` - strictly speaking. If we are really trying to be picky, I guess we could also remove `//model`, because it implicitly attaches to a link via `//model/@canonical_link`.

Personally, I'm happy either way as long as it is clear that any chain must terminate in a `//link` or `//world`.

Reply:
Ah, I see this is specifically following `@attached_to` attributes, not necessarily the frame attachments in the frame graph. Okay, we can leave `//joint` here.